### PR TITLE
[fix](scanner_schedule) scanner hangs due to negative num_running_scanners

### PR DESCRIPTION
**Pull request by dataroaring**
Before the patch, num_running_scanners is increased after submitting, then it may be decreased before increasing then negative values can be seen by get_block_from_queue and a expected submit does not happend.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

File: be/src/vec/exec/scan/scanner_context.cpp [hunk 1]
```diff
@@ -217,7 +219,14 @@ Status ScannerContext::get_block_from_queue(RuntimeState* state, vectorized::Blo
     // (if the scheduler continues to schedule, it will cause a lot of busy running).
     // At this point, consumers are required to trigger new scheduling to ensure that
     // data can be continuously fetched.
+    int64_t cur_bytes_in_queue = _cur_bytes_in_queue;
+    int32_t serving_blocks_num = _serving_blocks_num;
+    bool to_be_schedule = should_be_scheduled();
+    int num_running_scanners = _num_running_scanners;
+
+    bool is_scheduled = false;
     if (should_be_scheduled() && _num_running_scanners == 0) {
```
Comment (morningman) [inline be/src/vec/exec/scan/scanner_context.cpp:228]:
```suggestion
    if (to_be_schedule && _num_running_scanners == 0) {
```

**Comment by dataroaring** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Review by wm1581066 — APPROVED** [review]
LGTM

**Pull request merged**